app/models/content_item.rb

```diff
@@ -43,9 +45,8 @@ def add_taxonomies_by_id(taxons)
 private
 
   def linked_content(link_type)
-    search = Search.new
-    search.filter_by(link_type: link_type, source_ids: content_id)
-    search.execute
-    search.content_items
+    links = Link.where(link_type: link_type, source_content_id: content_id)
+
+    ContentItem.where(content_id: links.select(:target_content_id))
```

Comment:
Happy to talk about it. But it feels a bit better to me if the Domain objects use the ActiveRecord queries.